tools/sklearn/model_validation.xml

```diff
@@ -0,0 +1,221 @@
+            </when>
+            <when value="learning_curve">
+                <expand macro="feature_selection_estimator" />
+                <conditional name="extra_estimator">
+                    <expand macro="feature_selection_extra_estimator" />
+                    <expand macro="feature_selection_estimator_choices" />
+                </conditional>
+                <section name="options" title="Other Options" expanded="false">
+                    <!--groups-->
+                    <expand macro="model_validation_common_options"/>
+                    <!--train_sizes-->
+                    <expand macro="scoring"/>
+                    <param argument="exploit_incremental_learning" type="boolean" optional="true" truevalue="booltrue" falsevalue="boolfalse" checked="false" label="exploit_incremental_learning" help="Whether to apply incremental learning to speed up fitting of the estimator if supported"/>
+                    <expand macro="pre_dispatch"/>
+                    <expand macro="shuffle" checked="false" label="shuffle" help="Whether to shuffle training data before taking prefixes"/>
+                    <expand macro="random_state"/>
+                </section>
+            </when>
+            <when value="permutation_test_score">
+                <expand macro="feature_selection_estimator" />
+                <conditional name="extra_estimator">
+                    <expand macro="feature_selection_extra_estimator" />
+                    <expand macro="feature_selection_estimator_choices" />
+                </conditional>
+                <section name="options" title="Other Options" expanded="false">
+                    <!--groups-->
+                    <expand macro="model_validation_common_options"/>
+                    <expand macro="scoring"/>
+                    <param name="n_permutations" type="integer" value="" optional="true" label="n_permutations" help="Number of times to permute y"/>
+                    <expand macro="random_state"/>
+                </section>
+            </when>
+            <when value="validation_curve">
+                <expand macro="feature_selection_estimator" />
+                <conditional name="extra_estimator">
+                    <expand macro="feature_selection_extra_estimator" />
+                    <expand macro="feature_selection_estimator_choices" />
+                </conditional>
+                <section name="options" title="Other Options" expanded="false">
+                    <!--param_name-->
+                    <!--param_range-->
+                    <!--groups-->                    
+                    <expand macro="model_validation_common_options"/>
+                    <expand macro="scoring"/>
+                    <expand macro="pre_dispatch"/>
+                </section>
+            </when>
+        </conditional>
+        <expand macro="sl_mixed_input"/>
+    </inputs>
+    <outputs>
+        <data format="txt" name="outfile"/>
```

Comment:
the format can the `tabular` I think.

Reply:
Could you please give the details? Thanks.

Reply:
The output file is a tabular isn't it? Its tab saparated.

Reply:
Oh, yes. I changed. I thought the tabular datatype could solve the diff problem as well.